soc/nordic/nrf71/CMakeLists.txt

```diff
@@ -22,3 +22,54 @@ if(NOT CONFIG_BUILD_WITH_TFM AND CONFIG_PSA_NEED_CRACEN_KMU_DRIVER)
   # We are sure that this file will be included first.
   zephyr_linker_sources(RAM_SECTIONS SORT_KEY ! kmu_push_area_section.ld)
 endif()
+
+if(CONFIG_NRF_WIFI_WICR_GENERATE_HEX)
+  set(WICR_GEN_SCRIPT ${ZEPHYR_NRF_MODULE_DIR}/soc/nordic/nrf71/wicr/gen_wicr.py)
+  set(WICR_VALIDATE_SCRIPT ${ZEPHYR_NRF_MODULE_DIR}/soc/nordic/nrf71/wicr/validate_wicr.py)
+  set(WICR_HEX_OUTPUT ${ZEPHYR_BINARY_DIR}/wicr.hex)
+  set(ZEPHYR_DTS_FILE ${ZEPHYR_BINARY_DIR}/zephyr.dts)
+  set(ZEPHYR_HEX_FILE ${ZEPHYR_BINARY_DIR}/zephyr.hex)
+
+  if(CMAKE_VERBOSE_MAKEFILE)
+    set(WICR_ARGS_VERBOSE "--verbose")
+  else()
+    set(WICR_ARGS_VERBOSE)
+  endif()
+
+  # Validate WICR and IPC memory consistency after DTS is compiled
+  add_custom_target(wicr_validate
+    COMMAND ${PYTHON_EXECUTABLE} ${WICR_VALIDATE_SCRIPT}
+      --dts ${ZEPHYR_DTS_FILE}
+      ${WICR_ARGS_VERBOSE}
+    DEPENDS ${ZEPHYR_DTS_FILE}
+    WORKING_DIRECTORY ${ZEPHYR_BINARY_DIR}
+    COMMENT "Validating WICR and IPC memory consistency"
+    VERBATIM
+  )
+
+  # Generate wicr.hex after DTS is validated and zephyr.hex is built
+  add_custom_command(
```

Comment:
add this as a semi-zephyr project which sysbuild calls, see https://github.com/zephyrproject-rtos/zephyr/tree/main/soc/nordic/common/uicr for an example